policies/policies.go

```diff
@@ -162,4 +165,7 @@ type Repository interface {
 
 	// DeleteSinkFromAllDatasets removes a sink from a dataset
 	DeleteSinkFromAllDatasets(ctx context.Context, sinkID string, ownerID string) ([]Dataset, error)
+
+	// DeleteAGroupFromAllDatasets removes agent group from a dataset
+	DeleteAGroupFromAllDatasets(ctx context.Context, aGroup string, ownerID string) error
```

Comment:
again consistency: we shouldn't use "AgentGroup" in one place and "AGroup" in another